python/bigquery_handler.py

```diff
@@ -0,0 +1,136 @@
+import gzip
+
+from google.cloud import bigquery
+import json
+import os
+import re
+
+from plbart_relevant_code import preprocess
+
+project_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__))).replace("\\", "/") + '/'
+count_in_method = 0
+count_end_method = 0
+all_methods_dictionary = {}
+
+
+def tokenize_bigquery_json():
+    global all_methods_dictionary
+    filepath = project_path + 'bigquery_files/bq-results-20220207-151512-biubj4ywzbd1.json'
+    # filepath = project_path + 'bigquery_files/test.json'
+    data = []
+    with open(filepath, "r+", encoding='utf8') as file:
+        line = file.readline()
+        while line:
+            data.append(json.loads(line))
+            line = file.readline()
+    for instance in data:
+        assert not instance['binary']
+        single_file_content_methods_dictionary = go_through_content(instance['content'])
+        additions_to_values = {'path': instance['sample_path']}
+        for value in single_file_content_methods_dictionary.values():
+            value.update(additions_to_values)
+        all_methods_dictionary.update(single_file_content_methods_dictionary)
+
+    write_all_methods_in_dictionary_to_json()
+
+
+def go_through_content(content):
+    global count_in_method
+    global count_end_method
+    code_by_lines = iter(content.splitlines())
+    in_method = False
+    methods_dictionary = {}
+    brackets_counter = 0
+    comments = []
+    for line in code_by_lines:
+        if brackets_counter == 0 and is_method_beginning(line):
+            in_method = True
+            current_method_code_lines = []
+            count_in_method += 1
+            name = line.split("(")[0].split(" ")[-1]
+            number_of_words_in_name, name_with_spaces = get_number_of_words(name)
+            url_exists = False
+        if in_method:
+            current_method_code_lines.append(line)
+            brackets_counter += (line.count("{") - line.count("}"))
+            if not url_exists:
+                url_exists = bool(
+                    re.search(r'\.(get|navigate\(\).to)\([\w\s\+\\/]*(\"www|\"http|url|Url)[\"\w\s\+\\/]*\);', line))
+            if brackets_counter == 0:
+                # finish reading test
+                in_method = False
+                count_end_method += 1
+
+                if url_exists and number_of_words_in_name >= 5:
+                    # filtering methods
+                    instance = {'repo_name': name, 'name for description': name_with_spaces,
+                                'comments': ' '.join(comments), 'url': url_exists,
+                                'content': ' '.join(current_method_code_lines)}
+                    methods_dictionary[count_end_method] = instance
```

Comment:
currently count_end_method is the key in all_methods_dictionary, but it isn't getting written into the json file.
I was planning on writing the keys to the json file, so that after running the plbart code, it will be possible to retrieve the methods' descriptions.